pkg/networkpath/traceroute/runner.go

```diff
@@ -0,0 +1,217 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package traceroute
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sort"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/Datadog/dublin-traceroute/go/dublintraceroute/probes/probev4"
+	"github.com/Datadog/dublin-traceroute/go/dublintraceroute/results"
+
+	"github.com/google/uuid"
+)
+
+// TODO: are these good defaults?
+const (
+	DefaultSourcePort   = 12345
+	DefaultDestPort     = 33434
+	DefaultNumPaths     = 1
+	DefaultMinTTL       = 1
+	DefaultMaxTTL       = 30
+	DefaultDelay        = 50 //msec
+	DefaultReadTimeout  = 3 * time.Second
+	DefaultOutputFormat = "json"
+)
+
+// RunTraceroute wraps the implementation of traceroute
+// so it can be called from the different OS implementations
+//
+// This code is experimental and will be replaced with a more
+// complete implementation.
+func RunTraceroute(cfg Config) (NetworkPath, error) {
+	rawDest := cfg.DestHostname
+	dests, err := net.LookupIP(rawDest)
```

Comment:
Noticed that we can get this error for destination hostnames like app.datadoghq.com:
```
Error running check: failed to trace path: traceroute run failed: Invalid IPv4 address
```

It's because `net.LookupIP(rawDest)` sometimes return IPv6 Addresses. Something to address in future impl.

Shall we have this quick fix like this:
```suggestion
	dests, err := net.DefaultResolver.LookupIP(context.Background(), "ip4", rawDest)
```

This forces to IPv4.